SpatialGDK/Source/SpatialGDKServices/Private/SpatialPackageManager.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Async/Async.h"
+#include "SpatialPackageManager.h"
+#include "SpatialGDKServicesConstants.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialPackageManager);
+
+FSpatialPackageManager::FSpatialPackageManager() {}
+
+void FSpatialPackageManager::TryFetchRuntimeBinary(FString RuntimeVersion)
+{
+	FString RuntimePath = FPaths::Combine(SpatialGDKServicesConstants::GDKProgramPath, TEXT("runtime"), RuntimeVersion);
+
+	// Check if the binary already exists for a given version
+	if (FPaths::FileExists(FPaths::Combine(RuntimePath, SpatialGDKServicesConstants::RuntimeExe)))
+	{
+		UE_LOG(LogSpatialPackageManager, Verbose, TEXT("Runtime binary already exist."));
+		return;
+	}
+
+	// If it does not exist then fetch the binary using `spatial worker package retrieve`
+	UE_LOG(LogSpatialPackageManager, Log, TEXT("Trying to fetch runtime version %s"), *RuntimeVersion);
+
+	// Download and unzip the executable to "UnrealGDK\SpatialGDK\Binaries\ThirdParty\Improbable\Programs\Runtime\*version*"
+	FString Params =
+		FString::Printf(TEXT("package retrieve runtime %s %s %s --unzip"), *SpatialGDKServicesConstants::PlatformVersion, *RuntimeVersion, *RuntimePath);
+	FSpatialPackageManager FSpatialPackageManager = {};
```

Comment:
Not 100% sure of our convention but I think you should be alright to remove the  = {}